### PR TITLE
Added Hit FX, flash fx material, shake screen effect

### PR DESCRIPTION
**Pull request by thienphuochoang**

### CODE REVIEW

**Review by Betadrone — APPROVED** [review]

**Review by NoahMKuzyk — APPROVED** [review]
Screen shake SFX possibly bugged; flash and such works well. Pull from main and check if functionality was intended for screen shake.

**Review by NoahMKuzyk — COMMENTED** [review]
FX not SFX

File: CapstoneProject/Assets/Scripts/Core/Effect/CharacterEffect.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = System.Random;
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Effect/CharacterEffect.cs:5]: I've never seen an = in a using before, interesting. Does this change anything?

File: CapstoneProject/Assets/Scripts/Core/Effect/CharacterEffect.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            new Vector3(_shakePower.x * shakingDirection, _shakePower.y) * _shakeMultiplier;
+        _screenShake.GenerateImpulse();
+    }
+
+    public void GenerateHitFX(Transform targetTransform)
+    {
+        float xPosition = UnityEngine.Random.Range(-0.5f, 0.5f);
+        float yPosition = UnityEngine.Random.Range(-0.5f, 0.5f);
+        float zRotation = UnityEngine.Random.Range(-90, 90);
+        int whichHitFX = UnityEngine.Random.Range(0, 10);
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Effect/CharacterEffect.cs:79]: 10 here maybe could be a const for number of effects? is that was 10 represents here?

**Pull request merged**